src/actions/interfaces/action.interface.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Organization,
+  OrganizationResponse,
+} from '../../organizations/interfaces';
+import {
+  Invitation,
+  InvitationResponse,
+  OrganizationMembership,
+  OrganizationMembershipResponse,
+  User,
+  UserResponse,
+} from '../../user-management/interfaces';
+
+interface AuthenticationAction {
+  id: string;
+  object: 'authentication_action_context';
+  user: User;
+  organization?: Organization;
+  organizationMembership?: OrganizationMembership;
+  ipAddress?: string;
+  userAgent?: string;
+  issuer?: string;
+}
+
+export interface UserData {
+  object: 'user_data';
+  email: string;
+  firstName: string;
+  lastName: string;
+}
+
+interface UserRegistrationAction {
+  id: string;
+  object: 'user_registration_action_context';
+  userData: UserData;
+  invitation?: Invitation;
+  ipAddress?: string;
+  userAgent?: string;
+}
+
+export type Action = AuthenticationAction | UserRegistrationAction;
+
+interface AuthenticationActionPayload {
```

Comment:
Looks like a duplicate of `AuthenticationAction`

Reply:
This is the type that comes from the API response and it's snake cased. `AuthenticationAction` is what's returned to the developer and is camel cased.